src/slurm_plugin/slurm_resources.py

```diff
@@ -148,21 +148,25 @@ def is_drained(self):
 
         drained(sinfo) is equivalent to IDLE+DRAIN(scontrol) or DOWN+DRAIN(scontrol)
         """
-        return self._is_drain() and (
-            self.SLURM_SCONTROL_IDLE_STATE in self.state or self.SLURM_SCONTROL_DOWN_STATE in self.state
-        )
+        return self._is_drain() and (self.SLURM_SCONTROL_IDLE_STATE in self.state or self.is_down())
 
     def is_powering_down(self):
         """Check if slurm node is in powering down state."""
-        return self.SLURM_SCONTROL_POWERING_DOWN_STATE in self.state
+        return any(
+            powering_down_state in self.state for powering_down_state in self.SLURM_SCONTROL_POWERING_DOWN_STATES
+        )
 
     def is_power(self):
         """Check if slurm node is in power state."""
-        return self.SLURM_SCONTROL_POWER_STATE == self.state
+        states = self.state.split("+")
+        return all(power_state in states for power_state in self.SLURM_SCONTROL_POWER_STATE.split("+"))
 
     def is_down(self):
         """Check if slurm node is in a down state."""
-        return self.SLURM_SCONTROL_DOWN_STATE in self.state and not self.is_powering_down()
+        states = self.state.split("+")
```

Comment:
shall we do this split just once in the constructor and assign the result to a set? we could call the string `state_string` and the set `states`? up to you.

Also if you want in some other places like (is_online) you could check the state in the set rather than string which is more performant.

Reply:
Good idea! Add `self.state_string` and self.states to `SlurmNode`, change other places use `in self.state_string` to `in self.states`